src/ol3-layerswitcher.js

```diff
@@ -175,13 +184,24 @@
 
         var lyrTitle = lyr.get('title');
         var lyrId = ol.control.LayerSwitcher.uuid();
+        lyr.S.id = lyrId;
```

Comment:
We can't use the `S` property of the layer as it's internal to OpenLayers. I think the following is equivalent:

    lyr.set('id', lyrId);

It's making me wonder if custom properties added by the layer switcher should be prefixed to avoid them clashing with official properties in OpenLayers in future. Something like:

    lyr.set('LayerSwitcher:id', lyrId);